tekton/ci/jobs/tekton-catlin-lint.yaml

```diff
@@ -0,0 +1,78 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: catlin-lint
+  namespace: tektonci
+spec:
+  description: >-
+    This task detects changes in the pull request using git diff
+    and stores the changed tasks names in a workspace and passes
+    that to the catlin so that catlin can validate only changed
+    files.
+  workspaces:
+    - name: store-changed-files
+      description: Stores list of changed tasks
+  params:
+    - name: gitCloneDepth
+      description: Number of commits in the change + 1
+  resources:
+    inputs:
+      - name: source
+        type: git
+  steps:
+    - name: find-changed-tasks
+      image: docker.io/alpine/git:v2.26.2@sha256:23618034b0be9205d9cc0846eb711b12ba4c9b468efdd8a59aac1d7b1a23363f
+      workingDir: $(resources.inputs.source.path)
+      script: |
+        function detect_new_changed_tasks() {
+          # detect for changes in the task manifest
+          git --no-pager diff --name-only HEAD~$(( $(params.gitCloneDepth) - 1 ))|grep 'task/[^\/]*/[^\/]*/*[^/]*.yaml'|xargs -I {} dirname {}
+        }
+        all_tests=$(detect_new_changed_tasks |sort -u || true)
+        echo -n $all_tests > $(workspaces.store-changed-files.path)/changed-files.txt
+    - name: lint-catalog
+      image: gcr.io/tekton-releases/dogfooding/catlin:latest
```

Comment:
It looks good to only validate changed files. 
As a follow-up it might be good to have a way to run the linter on all files too.
If we use a specific sha here, instead of latest, we can detect when the image version is changed and run the linter on all resources in that case.

Reply:
> It looks good to only validate changed files.
> As a follow-up it might be good to have a way to run the linter on all files too.
> If we use a specific sha here, instead of latest, we can detect when the image version is changed and run the linter on all resources in that case.

yeah that would be a good idea. How about adding a nightly trigger pipeline (although I need to take a look how I can make that happen :sweat_smile: ) which will run `catlin` on all the resources?